include/fi_util.h

```diff
@@ -174,12 +175,15 @@ struct util_cq {
 	struct slist		err_list;
 	fi_cq_read_func		read_entry;
 	int			internal_wait;
+	fi_cq_progress_func	progress;
+	void			*arg;
 };
 
-int util_cq_open(const struct fi_provider *prov,
-		 struct fid_domain *domain, struct fi_cq_attr *attr,
-		 struct fid_cq **cq_fid, void *context);
-
+int util_cq_init(const struct fi_provider *prov, struct fid_domain *domain,
+		 struct fi_cq_attr *attr, struct util_cq *cq,
+		 fi_cq_progress_func progress, void *arg, void *context);
+void ofi_cq_progress(void *arg);
```

Comment:
Why not take struct util_cq * instead of void *arg?  A user can either cast util_cq back to their structure or obtain arg from util_cq, if a cast cannot be done.